Avoid duplication between 64k and 1MB linker scripts

0xTJ opened the pull request:
The main purpose of this PR is to split common firmware linker script parts out from the ROM-specific parts (and make it a bit prettier). This does make the linker scripts a bit bigger (though IMHO simpler), so I'm very open to thoughts on it.

In the ROM-specific linker script, the is a `MEMORY` statement for the board's address space, as well as one that specifies the (minimum) installed memory size. This is split apart to differentiate between the total ROM memory space (including mirrors) and the actual space available in the ROM.

In the common linker script (`INCLUDE` from the ROM-specific linker script), I've there is a `MEMORY` for the memory areas defined in the interface reference.

I've also cleaned up the `ALIGN` statements, letting sections provide the minimum required alignment. This required dropping the assumption that `.data` is at `_code_end` in ROM, since there might be alignment padding now. Instead, a `_data_load_start` symbol is provided. The vasm flag `-align` is also added to make sure that data statements will implicitly align (assembly instructions and C code/data is already aligned in the object sections).

I've copied the style of changes over the r68k, but wasn't sure if I should `INCLUDE` the common file from it, so I kept it as a single file.

This has been tested with 68010 on an r2  and an r1.2 with BIGROM (actually using a HUGEROM adapter, but the memory space is kept as BIGROM and the BIGROM image is repeated across the entire flash chips).

0xTJ committed:
Extend linker scripts to define all regions

0xTJ committed:
Put common ld script parts in a common file

0xTJ committed:
Move ld output format out of common

0xTJ committed:
Tweak stage2 ld script

0xTJ committed:
Cleanup alignment directives

0xTJ committed:
Add missing section directive

0xTJ committed:
Fix incorrect EFPT offset in ld script

0xTJ committed:
Fix vasm not providing natural alignment

0xTJ committed:
Make zip unsigned char
This has no effect, but figured it made sense for predictability.

0xTJ committed:
Propagate subset of changes to r68k

0xTJ committed:
Don't assume that .data LMA is right after text

0xTJ committed:
Update copyright years

0xTJ committed:
Round updateflash half_size up
Avoid missing the last byte when programming an odd-size image

roscopeco reviewed (dismissed):
Looks great, thanks @0xTJ !

0xTJ committed:
Fix a sign in linker script

Flipped a sign to correctly get the end of BASERAM for calculating UMEM length. This has no effect currently, since `ORIGIN(BASERAM)` is 0, but just in case it does this fixes it.

0xTJ commented:
> Looks great, thanks @0xTJ !

Thanks for the review! Before merging a did a quick check and noticed that a sign (on a 0-value, so no actual effect) was flipped. I added a commit to fix that, just in case it causes a problem in the future.

roscopeco reviewed (approved):
Looks good 👍
